src/service/Offers.ts

```diff
@@ -41,33 +40,47 @@ export const getOfferService = (
       id,
       instanceHandle,
       publicInvitationMaker,
-      proposalTemplate: { give, want },
+      proposalTemplate: { give: giveTemplate, want: wantTemplate },
     } = offer;
 
-    const mapPursePetnamesToBrands = paymentProposals =>
-      Object.fromEntries(
+    const convertProposals = async paymentProposals => {
+      const entries = await Promise.all(
         Object.entries(paymentProposals).map(
           // @ts-expect-error
-          ([kw, { pursePetname, value }]) => {
-            const brand = pursePetnameToBrand.get(pursePetname);
-            if (!brand) {
+          async ([kw, { pursePetname, value, amount: serializedAmount }]) => {
+            if (!serializedAmount && !pursePetnameToBrand.get(pursePetname)) {
               return [];
             }
-            return [
-              kw,
-              {
-                brand,
-                value: BigInt(value),
-              },
-            ];
+
+            /// XXX test e2e with dapp inter once feasible.
```

Comment:
thanks for the documenting. consider TODO.

Reply:
Done